returnn/dataset.py

```diff
@@ -149,3 +149,117 @@ def run(self):
                 hdf_writer.insert_batch(numpy.asarray([[speaker_index]], dtype="int32"), [1], [segment_name])
 
         hdf_writer.close()
+
+
+class ExtractSeqLensJob(Job):
+    """
+    Extracts sequence lengths from a dataset for one specific key.
+    """
+
+    def __init__(
+        self,
+        dataset: Dict[str, Any],
+        post_dataset: Optional[Dict[str, Any]] = None,
+        *,
+        key: str,
+        output_format: str,
+        returnn_config: Optional[ReturnnConfig] = None,
+        returnn_root: Optional[tk.Path] = None,
+    ):
+        """
+        :param dataset: dict for :func:`returnn.datasets.init_dataset`
+        :param post_dataset: extension of the dataset dict, which is not hashed
+        :param key: e.g. "data", "classes" or whatever the dataset provides
+        :param output_format: "py" or "txt".
+            "py" will write a Python dict seq_tag -> seq_len.
+            "txt" will write one seq_len per line.
+        :param returnn_config: for the RETURNN global config.
+            This is optional and only needed if you use any custom functions (e.g. audio pre_process)
+            which expect some configuration in the global config.
+        :param returnn_root: inserted to ``sys.path`` for the RETURNN import.
+        """
+        super().__init__()
+        self.dataset = dataset
+        self.post_dataset = post_dataset
+        self.key = key
+        assert output_format in {"py", "txt"}
+        self.output_format = output_format
+        self.returnn_config = returnn_config
+        self.returnn_root = returnn_root
+
+        self.out_returnn_config_file = self.output_path("returnn.config")
+        self.out_file = self.output_path(f"seq_lens.{output_format}")
+
+        self.rqmt = {"gpu": 0, "cpu": 1, "mem": 4, "time": 1}
+
+    @classmethod
+    def hash(cls, parsed_args):
+        """hash"""
+        parsed_args = parsed_args.copy()
+        parsed_args.pop("post_dataset")
+        if not parsed_args["returnn_config"]:
```

Comment:
Maybe add this to `Job.__sis_hash_exclude__` for more clarity?

Reply:
I don't think that's really necessary here. In my opinion, it doesn't really make it more clear. It's anyway not strictly necessary that I do this here at all. I just found it a bit cleaner.

So, if people don't like this here, I can also just remove it.

Reply:
@Icemole hash_exclude is only if you add new parameters to an existing job, but never for parameters you do not want to hash at all.

Reply:
> hash_exclude is only if you add new parameters to an existing job, but never for parameters you do not want to hash at all.

That is true, but programmatically these two lines are equivalent to
```
__sis_hash_exclude__ = {"returnn_config": None}
```
(i.e. it is only not hashed, if it evaluates to False - which is exactly if it is `None`)

Which then raises the question why we have it at all. Hashing `None` here would be perfectly fine.

Reply:
Oh yes thats right, it should be removed completely as it has no effect at all.